cloud-gateway-service/src/test/java/org/zowe/apiml/cloudgatewayservice/service/CentralApimlInfoMapperTest.java

```diff
@@ -0,0 +1,143 @@
+
+
+import com.netflix.appinfo.InstanceInfo;
+import org.apache.groovy.util.Maps;
+import org.assertj.core.util.Sets;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.zowe.apiml.cloudgatewayservice.service.model.ApimlInfo;
+import org.zowe.apiml.cloudgatewayservice.service.model.CentralServiceInfo;
+import org.zowe.apiml.services.ServiceInfo;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@ExtendWith(MockitoExtension.class)
+class CentralApimlInfoMapperTest {
+
+    @InjectMocks
+    private CentralApimlInfoMapper centralApimlInfoMapper;
+
+    @Nested
+    class WhenParametersAreInvalid {
+        @Test
+        void shouldThrowNPEWhenApimlIdIsNull() {
+            List<ServiceInfo> serviceInfoList = new ArrayList<>();
+            assertThatThrownBy(() -> centralApimlInfoMapper.buildApimlServiceInfo(null, serviceInfoList))
+                    .isExactlyInstanceOf(NullPointerException.class);
+        }
+
+        @Test
+        void shouldReturnEmptyServicesWhenServiceListIsNull() {
+            ApimlInfo info = centralApimlInfoMapper.buildApimlServiceInfo("apiml1", null);
+
+            assertThat(info.getApimlId()).isEqualTo("apiml1");
+            assertThat(info.getServices()).isEmpty();
+        }
+
+        @Test
+        void shouldReturnEmptyServicesWhenServiceListContainsNulls() {
+            ApimlInfo info = centralApimlInfoMapper.buildApimlServiceInfo("apiml1", asList(null, null, null));
+
+            assertThat(info.getApimlId()).isEqualTo("apiml1");
+            assertThat(info.getServices()).isEmpty();
+        }
+    }
+
+    @Nested
+    class WhenMappingServiceInfo {
+
+        private ServiceInfo serviceOne;
+        private ServiceInfo.Apiml apiml;
+        private ServiceInfo.ApiInfoExtended apiInfo;
+
+        @BeforeEach
+        public void setUp() {
+            centralApimlInfoMapper.metadataKeysAllowList = Sets.set("zos.sysname", "zos.system", "zos.sysplex", "zos.cpcName", "zos.zosName", "zos.lpar");
+
+            apiml = new ServiceInfo.Apiml();
+            apiInfo = ServiceInfo.ApiInfoExtended.builder().apiId("zowe.apiml.apicatalog").build();
+            apiml.setApiInfo(singletonList(apiInfo));
+            ServiceInfo.Instances instance = ServiceInfo.Instances.builder()
+                    .customMetadata(Maps.of("zos.sysname", "MD20",
+                            "zos.sysplex", "PLEXM2",
+                            "zos.jobid", "TST71112"
+                    )).build();
+
+            //Well decorated service
```

Comment:
is this comment needed?

Reply:
yes, I left a hit that if in future new fields will be add - they should be tested with this object, not minimal